docs/theory/diffusion/diffusion.rst

```diff
@@ -0,0 +1,713 @@
+        and then estimates diffusion constants for all the tracks.
+
+        Parameters
+        ----------
+        simulation_function : callable
+            Simulation function. Takes an SNR and produces simulations and a true diffusion constant.
+        snrs : array_like
+            SNRs to simulate for
+        methods : dict of callable
+            Dictionary with methods to apply to the tracks. Each dictionary
+            value should be a callable that takes `KymoTracks` and returns a list
+            of diffusion estimates.
+        """
+        results = {key: [] for key in methods.keys()}
+        var_estimates = {key: [] for key in methods.keys()}
+
+        for snr in snrs:
+            # Simulate our tracks for different snrs.
+            tracks, true_parameter = simulate_tracks(snr)
+
+            for method, estimates in results.items():
+                # Estimate constant for each track
+                diffusion_ests = methods[method](tracks)
+
+                # Extract diffusion constants and divide by the true value
+                estimates.append([est.value / true_parameter for est in diffusion_ests])
+                var_estimates[method].append([est.std_err**2 / true_parameter**2 for est in diffusion_ests])
+
+        return results, var_estimates
+
+This function generates a list of the estimates for each SNR for each method.
+To perform an analysis on the accuracy of these estimators, we need to define a simulation function and a dictionary with methods to apply to the tracks.
+Let's start by comparing the single track performance of the CVE, GLS and OLS method::
+
+    methods = {
+        "cve": lambda tracks: tracks.estimate_diffusion("cve"),
+        "gls": lambda tracks: tracks.estimate_diffusion("gls"),
+        "ols": lambda tracks: tracks.estimate_diffusion("ols"),
+    }
+
+    def simulate_tracks(snr):
+        """Function used to simulate tracks"""
+        params = snr_to_diffusion_parameters(snr)  # Determine diffusion parameters
+        return lk.simulation.simulate_diffusive_tracks(**params, num_tracks=500, steps=40), params["diffusion_constant"]
+
+    snrs = 10 ** np.arange(-1, 1.1, 0.25)
+    results, variances = test_estimation_methods(simulate_tracks, snrs, methods)
+
+Let's plot the two extremes in terms of SNR::
+
+    plt.figure()
+    ax1 = plt.subplot(2, 1, 1)
+    plt.hist(results["cve"][0], 30, label="cve");
+    plt.hist(results["ols"][0], 30, label="ols");
+    plt.title(f"SNR = {snrs[0]}")
+    plt.ylabel('Probability density')
+    plt.legend()
+    plt.subplot(2, 1, 2, sharex=ax1)
+    plt.hist(results["cve"][-1], 30, label="cve");
+    plt.hist(results["ols"][-1], 30, label="ols");
+    plt.title(f"SNR = {snrs[-1]}")
+    plt.ylabel('Probability density')
+    plt.xlabel('Diffusion constant [$\mu$m/s]')
+    plt.xlim([-20, 20])
+    plt.tight_layout()
+
+.. image:: figures/histogram_diff.png
+
```

Comment:
Detail: In the figure histogram_diff with high SNR, it would be nice to have CVE on top.